### PR TITLE
Refactor conversation components for improved structure and functiona…

### PR DESCRIPTION
**Pull request by hasith**
…lity: Introduced ParticipantMenuBar and ParticipantMenuButton for better menu handling in conversations. Added sanitizeTopicDisplayName utility for safe topic display. Updated layout and comments for clarity and robustness in handling participant roles and themes.

### CODE REVIEW

File: src/app/(dashboard)/conversations/utils/index.ts [hunk 1]
```diff
@@ -14,6 +14,27 @@ export function getTopicDisplayName(topicId: string): string {
   return topicId === 'general-discussions' ? 'General Discussions' : topicId;
 }
 
+/**
+ * Sanitize a URL-derived topic param for safe display in the UI.
+ * Used when creating synthetic topics from query params (e.g. ?topic=...) —
+ * raw values can be malformed or contain path-like strings.
+ */
```
Comment (xiana-agent) [inline src/app/(dashboard)/conversations/utils/index.ts]: **Warning:** `sanitizeTopicDisplayName` processes user-controlled URL input and has no unit tests. The regex logic (control char stripping, `decodeURIComponent` fallback, 200-char truncation) is exactly the kind of code that benefits from edge-case coverage. Consider adding tests for: `null`, empty string, control characters (`\x00`, `\x1F`), strings exceeding 200 chars, invalid percent-encoded sequences (`%zz`), and `+`-encoded spaces.

File: src/app/(dashboard)/conversations/[agentName]/[activationName]/_components/participant-menu-bar.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+'use client';
+
+import { PanelLeft } from 'lucide-react';
+
```
Comment (xiana-agent) [inline src/app/(dashboard)/conversations/[agentName]/[activationName]/_components/participant-menu-bar.tsx]: **Warning:** `ParticipantMenuButton` is declared here without `export`, then re-exported at the bottom of the file via `export { ParticipantMenuButton }`. `ParticipantMenuBar` on the same file uses `export function` at declaration. For consistency, prefer `export function ParticipantMenuButton` here and remove the re-export at line 35.

File: src/app/(dashboard)/conversations/[agentName]/[activationName]/page.tsx [hunk 3]
```diff
@@ -278,9 +279,10 @@ function ConversationContent() {
       } else if (!topicExists) {
```
Comment (xiana-agent) [inline src/app/(dashboard)/conversations/[agentName]/[activationName]/page.tsx]: **Suggestion:** The `label` prop passed to `ParticipantMenuBar` further down uses `activationName || agentName || 'Agent'` — raw decoded URL params. Since `sanitizeTopicDisplayName` was introduced for exactly this pattern, consider applying it to the label for consistency: `sanitizeTopicDisplayName(activationName || agentName) || 'Agent'`. JSX prevents XSS regardless, but this keeps the defensive approach consistent across the file.
